test/e2e/xtx/manager.go

```diff
@@ -0,0 +1,99 @@
+package xtx
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/big"
+	"strings"
+
+	"github.com/omni-network/omni/contracts/bindings"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/test/e2e/netman"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type TxSenderManager struct {
+	txSender map[uint64]Sender
+	abi      *abi.ABI
```

Comment:
probably rename to `portalABI` since you will need multiple instances, once for each contract that you support sending/calling to.